### PR TITLE
Add remove functionality

### PR DESCRIPTION
**Pull request by Estherstarr**
To-Do List: Add & Remove

- [ ] Remove all hardcoded items from the tasks array.
- [ ] Create a new JavaScript file for the new functionality.
- [ ] Implement a function for adding a new task (add a new element to the array).
- [ ] Implement a function for deleting a task 
- This can be achieved by clicking inside the taskbar
- [ ] Implement a function for editing task descriptions.
- [ ] By default new tasks should have the property completed set to false and the property index set to the value of the new array length.
- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- [ ] All changes to the To-Do List should be saved in local storage.

Screenshot 2022-04-01 at 5 31 24 AM
Screenshot 2022-04-01 at 5 34 17 AM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+
+### Setup
+Git clone - https://github.com/esterstarr/My-Portfolio
+HTML, CSS Linter configuration - https://github.com/microverseinc/linters-config/tree/master/html-css
+
+### Deployment
+Github
+
+
+👤 **Author1**
```
Comment (rloterh) [inline README.md]: - [x] OPTIONAL
Kindly replace the text **Author1** with your full name, so as to make your ReadMe more professional

File: src/index.js [hunk 2]
```diff
@@ -1,38 +1,81 @@
...
-            <button id="three-dots" class="h5 btn m-0 icon">
-              <i class="fa-solid fa-ellipsis-vertical"></i>
-            </button>
-            <button id="trash" class="h5 btn m-0 icon hide">
-              <i class="fa-solid fa-trash-can"></i>
-            </button>
-          </div>
-        `;
+/* eslint-disable */
+/* eslint-enable */
```
Comment (rloterh) [inline src/index.js]: - [x] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean.

File: src/index.js [hunk 3]
```diff
@@ -1,38 +1,81 @@
...
+  e.preventDefault();
+  Task.TaskObject = [];
+  localStorage.setItem('TASKS_LIST', JSON.stringify(Task.TaskObject));
+  checkLocalStorage();
+});
+
+Elements.taskList.addEventListener('click', (e) => {
+  e.stopPropagation();
+
+  [...Elements.taskList.children].forEach((item, index) => {
```
Comment (rloterh) [inline src/index.js:31]: - [x] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal 5). Also, deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Kindly ensure to improve upon this aspect.

**Index mismatch:**
image

File: src/module/constElements.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const refreshTask = document.querySelector('#refresh');
+const taskInput = document.querySelector('#task');
+const submitInput = document.querySelector('#arrow-submit');
+const taskList = document.querySelector('#taskList');
+const clearCompletedTasks = document.querySelector('#clear-completed');
```
Comment (rloterh) [inline src/module/constElements.js:5]: - [x] OPTIONAL
Select checkboxes are not reflected in the local storage. Kindly ensure that these changes are stored in the local storage

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Onomeh, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (Estherstarr) [reply src/index.js:31]: Thank you @rloterh for your review
For this part the description property is used to match the value
Screenshot 2022-04-01 at 7 50 11 AM

I'm using the code you highlighted to change the color of the taskbar when clicked to yellow and also enable the delete functionality

File: src/module/constElements.js (hunk 4 above)
Comment (Estherstarr) [reply src/module/constElements.js:5]: Based on the project sequence, this feature will be enabled in the next project addition
Thank you for the review.

**Comment by Estherstarr** [comment]
The following changes were made:-
- [ ] Enable linter check on your code to ensure that your code is robust code and clean.
- [ ]  Kindly replace the text Author1 with your full name, so as to make your ReadMe more professional

File: src/index.js (hunk 3 above)
Comment (rloterh) [reply src/index.js:31]: @Estherstarr 
This requirement simply means that the first entry should have an index of 1, and the second an index of 2, and so on. Assuming the first element is deleted, then the second element should replace the first element and then it should have an index of 1. Cheers!

File: src/module/constElements.js (hunk 4 above)
Comment (rloterh) [reply src/module/constElements.js:5]: @Estherstarr 
Well noted 😃 🙏

File: src/index.js (hunk 3 above)
Comment (Estherstarr) [reply src/index.js:31]: @rloterh how do I enable this feature. Can you give me some guidelines

**Comment by Estherstarr** [comment]
- [ ] Update the first entry to have an index of 1
- [ ] Enable the function to update with the appropriate index

File: src/module/Task.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    localStorage.setItem('TASKS_LIST', JSON.stringify(this.TaskObject));
+  }
+
+  remove(element) {
+    if (element.classList.contains('taskDynamic')) {
+      element.remove();
+      this.TaskObject.splice(element.getAttribute('data-id'), 1);
+    }
+    localStorage.setItem('TASKS_LIST', JSON.stringify(this.TaskObject));
+  }
```
Comment (FrederickMih) [inline src/module/Task.js]: - [ ] I have done several attempts to delete a task but to no avail. Please make sure to fix this.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Estherstarr , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (rloterh) [reply src/index.js:31]: @Estherstarr 
We have had a discussion on the same via slack

File: src/module/Task.js (hunk 5 above)
Comment (rloterh) [reply src/module/Task.js:25]: @FrederickMih 
This project is complete!
Tasks are deleting effectively when I reviewed them.

Comment (FrederickMih) [reply src/module/Task.js]: @rloterh, I have tested your project over and over

Comment (Estherstarr) [reply src/module/Task.js]: Thank you @FrederickMih for the review.
I corrected the error, kindly review to enable me proceed

**Pull request closed without merging**